File: job-quest-backend/src/main/java/com/ott/job_quest_backend/model/Application.java
package com.ott.job_quest_backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Application {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private int id;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getRoleName() {
        return roleName;
    }

    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }

    public String getCompanyName() {
        return companyName;
    }

    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }

    public Date getDateOfApplication() {
        return dateOfApplication;
    }

    public void setDateOfApplication(Date dateOfApplication) {
        this.dateOfApplication = dateOfApplication;
    }

    public String getPlatform() {
        return platform;
    }

    public void setPlatform(String platform) {
        this.platform = platform;
    }

    public ApplicationStatus getStatus() {
        return status;
    }

    public void setStatus(ApplicationStatus status) {
        this.status = status;
    }

    public String getResume() {
        return resume;
    }

    public void setResume(String resume) {
        this.resume = resume;
    }

    public String getJobDescription() {
        return jobDescription;
    }


    public void setJobDescription(String jobDescription) {
        this.jobDescription = jobDescription;
    }

    public String getRemarks() {
        return remarks;
    }

    public void setRemarks(String remarks) {
        this.remarks = remarks;
    }

    public String getTag() {
        return tag;
    }

    public void setTag(String tag) {
        this.tag = tag;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    private String roleName;
    private String companyName;

    @Temporal(TemporalType.DATE)
    private Date dateOfApplication;

    private String platform;

    @Enumerated(EnumType.STRING)
    private ApplicationStatus status;

    @NotBlank
    private String resume;

    @Size(max=5000)
    @Column(length=5000)
    private String jobDescription;
    private String remarks;
    private String tag;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt = new Date();

    @Temporal(TemporalType.TIMESTAMP)
    private Date updatedAt = new Date();

    @ManyToOne
    @JoinColumn(name="user_id", nullable = false)
    @JsonIgnore
    private User user;

}

File: job-quest-client/src/app/models/application.ts
import { ApplicationStatus } from "../enums/application-status";

export class Application {
  id: number;
  roleName: string;
  companyName: string;
  tag: string;
  remarks: string;
  jobDescription: string;
  platform: string;
  status: ApplicationStatus;
  dateOfApplication: Date;
  resume: string;
  createdAt: Date;
  updatedAt: Date;


  constructor(
    roleName: string,
    companyName: string,
    tag: string,
    remarks: string,
    jobDescription: string,
    platform: string,
    status: ApplicationStatus,
    dateOfApplication: Date,
    resume: string,
    createdAt: Date,
    updatedAt: Date,
    id: number
  ) {
    this.roleName = roleName;
    this.companyName = companyName;
    this.tag = tag;
    this.remarks = remarks;
    this.jobDescription = jobDescription;
    this.platform = platform;
    this.status = status;
    this.dateOfApplication = dateOfApplication;
    this.resume = resume;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.id = id;
  }

}
